### PR TITLE
matrix-synapse.plugins.matrix-synapse-s3-storage-provider: add missin…

### PR DESCRIPTION
**Pull request by fabaff**
## Description of changes
Fix build (https://hydra.nixos.org/build/231674982)

https://github.com/NixOS/nixpkgs/pull/229192 was merged without testing if the build process passes, I assume.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Another issue:

```bash
error: builder for '/nix/store/wzgkq15lhiqxa0d7gzc0lc8xfh9sbcj0-python3.10-matrix-synapse-s3-storage-provider-1.2.1.drv' failed with exit code 1;
       last 10 log lines:
       >     return _bootstrap._gcd_import(name[level:], package, level)
       >   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
       >   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
       >   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
       >   File "/nix/store/vbmyzqn83fc01kh2ilbphmxhbypc3wxf-python3.10-matrix-synapse-s3-storage-provider-1.2.1/lib/python3.10/site-packages/s3_storage_provider.py", line 30, in <module>
       >     from synapse.logging.context import LoggingContext, make_deferred_yieldable
       > ModuleNotFoundError: No module named 'synapse'
       For full logs, run 'nix-store -l /nix/store/wzgkq15lhiqxa0d7gzc0lc8xfh9sbcj0-python3.10-matrix-synapse-s3-storage-provider-1.2.1.drv'.
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 249278` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider</li>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
@yu-re-ka, thanks for the hint about `matrix-synapse-unwrapped`.

File: pkgs/servers/matrix-synapse/plugins/s3-storage-provider.nix [hunk 1]
```diff
@@ -16,13 +30,26 @@ buildPythonPackage rec {
...
   '';
 
+  buildInputs = [
+    matrix-synapse-unwrapped
+  ];
+
+  propagatedBuildInputs = [
+    boto3
+    humanize
+    psycopg2
```
Comment (yu-re-ka) [inline pkgs/servers/matrix-synapse/plugins/s3-storage-provider.nix:40]: psycopg2 etc. are implied in `matrix-synapse-unwrapped.propagatedBuildInputs`. I would prefer if this stayed in sync with the exact inputs used by synapse

**Review by yu-re-ka — APPROVED** [review]

File: pkgs/servers/matrix-synapse/plugins/s3-storage-provider.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/servers/matrix-synapse/plugins/s3-storage-provider.nix]: Makes no sense to do it different with this plugin, change reverted.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 249278` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider</li>
    <li>matrix-synapse-plugins.matrix-synapse-s3-storage-provider.dist</li>
  </ul>
</details>

**Pull request merged**